macro/src/lib.rs

```diff
@@ -3,8 +3,8 @@
 // see LICENSE for license details.
 
 //! Subxt macro for generating Substrate runtime interfaces.
-
-extern crate proc_macro;
+// TODO: The workspace lint is not working properly so it's disabled here for now.
+#![allow(clippy::manual_unwrap_or_default)]
```

Comment:
Do we need this if you also are pulling lints from the workspace?

Reply:
It doesn't work without this.

I got:

```bash
  process didn't exit successfully: `/home/niklasad1/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/clippy-driver /home/niklasad1/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustc --crate-name subxt_macro --edition=2021 macro/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=273 --crate-type proc-macro --emit=dep-info,metadata -C prefer-dynamic -C embed-bitcode=no -C debuginfo=2 --deny=unused_parens --deny=unused_extern_crates --deny=unused_comparisons --deny=unused_allocation --deny=unconditional_recursion '--allow=clippy::type_complexity' --deny=patterns_in_fns_without_body --deny=path_statements --deny=overflowing_literals --deny=non_shorthand_field_patterns --deny=no_mangle_generic_items --deny=missing_docs '--allow=clippy::manual-unwrap-or-default' --deny=improper_ctypes --deny=bad_style '--deny=clippy::all' -C metadata=43244fc89d4ec22e -C extra-filename=-43244fc89d4ec22e --out-dir /home/niklasad1/Github/subxt/target/debug/deps -C incremental=/home/niklasad1/Github/subxt/target/debug/incremental -L dependency=/home/niklasad1/Github/subxt/target/debug/deps --extern darling=/home/niklasad1/Github/subxt/target/debug/deps/libdarling-8565ccfffaac534e.rmeta --extern codec=/home/niklasad1/Github/subxt/target/debug/deps/libparity_scale_codec-d4175ab4871a818f.rmeta --extern proc_macro_error=/home/niklasad1/Github/subxt/target/debug/deps/libproc_macro_error-3ca2393b60a689cf.rmeta --extern quote=/home/niklasad1/Github/subxt/target/debug/deps/libquote-123956ecf9e3264f.rmeta --extern scale_typegen=/home/niklasad1/Github/subxt/target/debug/deps/libscale_typegen-24a6bdebbb8b8927.rmeta --extern subxt_codegen=/home/niklasad1/Github/subxt/target/debug/deps/libsubxt_codegen-5eec60c5ec0c06a1.rmeta --extern syn=/home/niklasad1/Github/subxt/target/debug/deps/libsyn-c52220c64d6d07d1.rmeta --extern proc_macro -L native=/home/niklasad1/Github/subxt/target/debug/build/ring-fd1d525f9c796c1d/out` (exit status: 1)
```

So, `'--allow=clippy::manual-unwrap-or-default'` is actually passed when compiling the crate but fails anyway...
That's why this TODO exist :(

Reply:
Yup, this lint setting is not getting propagated for some reason